### PR TITLE
evm: Add a stub chainwriter impl

### PR DESCRIPTION
**Pull request by nickcorin**
Added a stub implementation for the `ChainWriterService` into the EVM relayer.

The `ChainWriter` interface does NOT belong here, and will be removed once [#523](https://github.com/smartcontractkit/chainlink-common/pull/523) has been merged. It has been copy-pasted here for review and build purposes.

### CODE REVIEW

File: core/services/relay/evm/chain_writer.go
Comment (jmank88) [inline core/services/relay/evm/chain_writer.go]: How about `w` for these method receivers:
```suggestion
func (w *chainWriter) SubmitSignedTransaction(ctx context.Context, payload []byte, signature map[string]any, transactionID uuid.UUID, toAddress string, meta *TxMeta, value big.Int) (int64, error) {
```

> For a method receiver, one or two letters is sufficient.

https://go.dev/wiki/CodeReviewComments#variable-names

File: core/services/relay/evm/chain_writer.go
Comment (ilija42) [inline core/services/relay/evm/chain_writer.go]: Why isn't the interface definition pulled from common?

Comment (nickcorin) [reply core/services/relay/evm/chain_writer.go]: It will be. If you see the description of the PR, this will be removed as soon as the PR in common is merged. I added the interface here temporarily so the build didn't break.

File: core/services/relay/evm/chain_writer.go
Comment (nickcorin) [reply core/services/relay/evm/chain_writer.go]: Done.

File: core/services/relay/evm/chain_writer.go
Comment (ilija42) [reply core/services/relay/evm/chain_writer.go]: Ah my bad didn't read the description. Although you can reference the PR in common in this PR before it is merged

File: core/services/ocr2/plugins/median/services.go
Comment (nickcorin) [inline core/services/ocr2/plugins/median/services.go]: This change is irrelevant to the `ChainWriter` work, however updating the `chainlink-common` dependency caught this error.

**Review by ilija42 — CHANGES_REQUESTED** [review]
There is a Sonar complaint 
`[File is not goimports -ed with -local github.com/smartcontractkit/chainlink]`
in chain_writer.go:9

**Review by ilija42 — DISMISSED** [review]

File: core/services/ocr2/plugins/median/services.go
Comment (jmank88) [reply core/services/ocr2/plugins/median/services.go]: Are you sure you aren't downgrading chainlink-common? IIRC This is a brand new argument

**Review by patrick-dowell — DISMISSED** [review]
LGTM

File: core/services/ocr2/plugins/median/services.go
Comment (nickcorin) [reply core/services/ocr2/plugins/median/services.go]: I did accidentally, yes. It's been added back.

**Review by jmank88 — DISMISSED** [review]

**Review by ilija42 — APPROVED** [review]

File: core/scripts/go.mod
Comment (ilija42) [inline core/scripts/go.mod:24]: @jmank88 Is it fine to have scripts and integration point to different hashes? Tidy doesn't do anything

File: .changeset/tricky-flowers-exist.md
Comment (jmank88) [inline .changeset/tricky-flowers-exist.md:5]: Let's make this #internal since it doesn't concern node ops

**Review by jmank88 — APPROVED** [review]

File: core/scripts/go.mod
Comment (jmank88) [reply core/scripts/go.mod:24]: If tidy is satisfied then no problem, but did you use the make command? Regular go mod tidy only does a single module

Comment (nickcorin) [reply core/scripts/go.mod:24]: Yeah I used the `make` command.

**Review by prashantkumar1982 — APPROVED** [review]

File: core/services/relay/evm/chain_writer.go
Comment (amit-momin) [inline core/services/relay/evm/chain_writer.go:17]: nit since you'll need to push a commit to fix the merge conflict anyways: Think we should group this import with the `commonservices` one above since both are from `chainlink-common`

Comment (amit-momin) [reply core/services/relay/evm/chain_writer.go:17]: Actually I forgot you'd maintain the approvals after resolving the merge conflicts so you can ignore this. Not important enough to try to get approvals again

**Pull request closed without merging**
